src/components/Report.tsx

```diff
@@ -1,15 +1,43 @@
 import React, { useState, useEffect } from "react";
-import { Col, Row, Container } from "reactstrap";
 import gradesData from "../grades.json";
 import gradeScaleData from "../utils/gradeScale.json";
-// import Logo from "../assets/unnamed.jpg";
-import Logo from "../assets/sept-school-logo.png";
+import attendanceData from "../utils/attendance.json";
+import Logo from "../assets/unnamed.jpg";
+import { calculateCredits } from "../utils/creditCalculator";
+import { rateAttendance, AttendanceRecord } from "../utils/attendanceRating";
+import courseDescriptions from "../utils/courseDescriptions";
+import { TOTAL_SESSIONS, MAX_CREDITS } from "../utils/config";
+import "../Report.css";
+
+// Constants for attendance indices
+const ABSENCES = attendanceData.attendanceCategories.indexOf("A"); // Index for 'A' (Absences)
```

Comment:
You have this and the following already defined in a utility file, reuse that ... great choice creating the utility for this :D